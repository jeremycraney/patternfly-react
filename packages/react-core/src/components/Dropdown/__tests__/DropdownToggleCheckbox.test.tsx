import * as React from 'react';
import { render } from '@testing-library/react';
import { mount } from 'enzyme';
import { DropdownToggleCheckbox } from '../DropdownToggleCheckbox';

const props = {
  onChange: jest.fn(),
  isChecked: false
};

test('controlled', () => {
  const view = render(<DropdownToggleCheckbox isChecked id="check" aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('uncontrolled', () => {
  const view = render(<DropdownToggleCheckbox id="check" aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('with text', () => {
  const view = render(
    <DropdownToggleCheckbox id="check" isDisabled aria-label="check">
      Some text
    </DropdownToggleCheckbox>
  );
  expect(view.container).toMatchSnapshot();
});

test('isDisabled', () => {
  const view = render(<DropdownToggleCheckbox id="check" isDisabled aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('3rd state', () => {
  const view = render(<DropdownToggleCheckbox id="check" isChecked={null} aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('passing class', () => {
  const view = render(
    <DropdownToggleCheckbox label="label" className="class-123" id="check" isChecked aria-label="check" />
  );
  expect(view.container).toMatchSnapshot();
});

test('passing HTML attribute', () => {
  const view = render(
    <DropdownToggleCheckbox label="label" aria-labelledby="labelId" id="check" isChecked aria-label="check" />
  );
  expect(view.container).toMatchSnapshot();
});

test('checkbox passes value and event to onChange handler', () => {
  const newValue = true;
  const event = {
    target: { checked: newValue }
  };
  const view = mount(<DropdownToggleCheckbox id="check" {...props} aria-label="check" />);
  view.find('input').simulate('change', event);
  expect(props.onChange.mock.calls[0][0]).toBe(newValue);
  expect(props.onChange.mock.calls[0][1]).toMatchObject(event);
});
