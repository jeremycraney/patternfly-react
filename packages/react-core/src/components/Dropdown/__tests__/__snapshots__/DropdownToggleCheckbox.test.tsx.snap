// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`3rd state 1`] = `
<div>
  <label
    class="pf-c-dropdown__toggle-check"
    for="check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      data-ouia-component-id="OUIA-Generated-DropdownToggleCheckbox-5"
      data-ouia-component-type="PF4/DropdownToggleCheckbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`controlled 1`] = `
<div>
  <label
    class="pf-c-dropdown__toggle-check"
    for="check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      checked=""
      data-ouia-component-id="OUIA-Generated-DropdownToggleCheckbox-1"
      data-ouia-component-type="PF4/DropdownToggleCheckbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`isDisabled 1`] = `
<div>
  <label
    class="pf-c-dropdown__toggle-check"
    for="check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      data-ouia-component-id="OUIA-Generated-DropdownToggleCheckbox-4"
      data-ouia-component-type="PF4/DropdownToggleCheckbox"
      data-ouia-safe="true"
      disabled=""
      id="check"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`passing HTML attribute 1`] = `
<div>
  <label
    class="pf-c-dropdown__toggle-check"
    for="check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      aria-labelledby="labelId"
      checked=""
      data-ouia-component-id="OUIA-Generated-DropdownToggleCheckbox-7"
      data-ouia-component-type="PF4/DropdownToggleCheckbox"
      data-ouia-safe="true"
      id="check"
      label="label"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`passing class 1`] = `
<div>
  <label
    class="pf-c-dropdown__toggle-check class-123"
    for="check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      checked=""
      data-ouia-component-id="OUIA-Generated-DropdownToggleCheckbox-6"
      data-ouia-component-type="PF4/DropdownToggleCheckbox"
      data-ouia-safe="true"
      id="check"
      label="label"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`uncontrolled 1`] = `
<div>
  <label
    class="pf-c-dropdown__toggle-check"
    for="check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      data-ouia-component-id="OUIA-Generated-DropdownToggleCheckbox-2"
      data-ouia-component-type="PF4/DropdownToggleCheckbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`with text 1`] = `
<div>
  <label
    class="pf-c-dropdown__toggle-check"
    for="check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      data-ouia-component-id="OUIA-Generated-DropdownToggleCheckbox-3"
      data-ouia-component-type="PF4/DropdownToggleCheckbox"
      data-ouia-safe="true"
      disabled=""
      id="check"
      type="checkbox"
    />
    <span
      aria-hidden="true"
      class="pf-c-dropdown__toggle-text"
      id="check-text"
    >
      Some text
    </span>
  </label>
</div>
`;
